recipes/cesm-pop-lowres-1deg/recipe.py

```diff
@@ -0,0 +1,44 @@
+from pangeo_forge_recipes.patterns import ConcatDim, FilePattern, MergeDim
+from pangeo_forge_recipes.recipes import XarrayZarrRecipe
+
+
+def make_full_path(variable, time, api_params=""):
+    """Returns a valid path to the source files
+    """
+    return (
+        f"https://tds.ucar.edu/thredds/fileServer/datazone/campaign/cesm/collections/ASD/"
+        f"v5_rel04_BC5_ne30_g16/ocn/proc/tseries/daily/v5_rel04_BC5_ne30_g16.pop.h.nday1."
+        f"{variable}.{time}.nc{api_params}"
+    )
+
+
+vars = [
+    "HMXL_2",
+    "SFWF_2",
+    "SHF_2",
+    "SSH_2",
+    "SSS",
+    "SST",
+    "SST2",
+    "TAUX_2",
+    "TAUY_2",
+    "U1_1",
+    "U2_2",
+    "V1_1",
+    "V2_2",
+    "XMXL_2",
+]
+
+concat_dim = ConcatDim("time", keys=["00010101-01661231"])
+
+merge_dim = MergeDim("variable", keys=vars)
+
+chunks = {"time": 200}
+
+
+def instantiate_recipe(api_params):
+
+    make_full_path.__defaults__ = (api_params,)
```

Comment:
I would really prefer if we could not use this code style in recipes. It's not a good practice to manipulate `__defaults__`. Whay do we have to do it?

Reply:
> Why do we have to do it?

That is an artifact of https://github.com/pangeo-forge/pangeo-forge-recipes/pull/167 being incomplete. Once I rewrite that PR following your suggestion in https://github.com/pangeo-forge/pangeo-forge-recipes/pull/167#issuecomment-882727677, this use of `__defaults__` will not be necessary.

I have leaned on this approach in other situations, e.g. [here](https://github.com/pangeo-forge/staged-recipes/pull/26/files#diff-a0aab5b377d945854507716ad6d82a5627da3b1586c9337a6edb26911d6538bcR31-R32). Now that I've gained more recipe writing experience, I can see there are other ways to handle that scenario, and will make a note rewrite that recipe and avoid manipulating `__defaults__` going forward.

Reply:
Noting that `__defaults__` manipulation was removed by https://github.com/pangeo-forge/staged-recipes/pull/56/commits/8c58c856910e1f72c986841fa20c5a58a769f218